infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -402,10 +477,10 @@ private long getServerEpoch() {
      *
      * @return the phase 2 rank
      */
-    public Rank getPhase2Rank() {
-        Phase2Data phase2Data = getPhase2Data();
-        if (phase2Data != null) {
-            return phase2Data.getRank();
+    public Rank getAcceptedRank() {
+        AcceptedData acceptedData = getAcceptedData();
```

Comment:
Optional?

Reply:
Let me deal with Optional is a separate patch. I tried doing it, but then I realized the change percolates to quite a bit of code.